From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Wed, 1 May 2024 10:47:36 +0200
Subject: layout-manager: Return min padding in non-device mode
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

If we don't want automatic layout based on device information we should
return the minimal padding rather than 0 as otherwise elements will be
too close to the screen edge.

While at that make it simpler to debug such situations.

Fixes: 474b07f90 ("layout-manager: Allow to query rounded corner information")

Signed-off-by: Guido Günther <agx@sigxcpu.org>
Part-of: <https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1424>
(cherry picked from commit 5c53008a12d508d125ba7fb58d70a95044a2a60c)
---
 src/layout-manager.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/layout-manager.c b/src/layout-manager.c
index 2b84652..8ccbd83 100644
--- a/src/layout-manager.c
+++ b/src/layout-manager.c
@@ -167,10 +167,11 @@ get_corner_shift (PhoshLayoutManager *self)
 {
   float r, a, b, c, scale;
   PhoshShellLayout layout;
+  guint shift = PHOSH_TOP_BAR_MIN_PADDING;
 
   layout = g_settings_get_enum (self->settings, SHELL_LAYOUT_KEY);
   if (layout != PHOSH_SHELL_LAYOUT_DEVICE)
-    return 0;
+    goto out;
 
   scale = phosh_monitor_get_fractional_scale (self->builtin);
   r = c = gm_display_panel_get_border_radius (self->panel) / scale;
@@ -190,7 +191,12 @@ get_corner_shift (PhoshLayoutManager *self)
   b = c - (PHOSH_TOP_BAR_HEIGHT - PHOSH_TOP_BAR_ICON_SIZE) / 2;
   a = floor (sqrt((c * c) - (b * b)));
 
-  return MAX (PHOSH_TOP_BAR_MIN_PADDING, ceil (r - a));
+  shift = MAX (PHOSH_TOP_BAR_MIN_PADDING, ceil (r - a));
+
+ out:
+  g_debug ("Corner shift: %u", shift);
+
+  return shift;
 }
 
 
